src/core/ActivityInfoTree.h

```diff
@@ -59,12 +61,17 @@ protected slots:
 private:
     // this is the full activity list, it never changes
     QList<ActivityInfo *> m_menuTreeFull;
+    QList<ActivityInfo *> m_menuTreeFulltemp;
     // represents the Menu view and can be filtered
     QList<ActivityInfo *> m_menuTree;
+    QList<ActivityInfo *> m_searched;
     ActivityInfo *m_rootMenu;
     ActivityInfo *m_currentActivity;
     static int menuTreeCount(QQmlListProperty<ActivityInfo> *property);
     static ActivityInfo *menuTreeAt(QQmlListProperty<ActivityInfo> *property, int index);
+    QObject *m_mainWindow;
+    int templength = 0;
+    QString searchedtext;
```

Comment:
for me we can still gain in readabily using standard algorithms.
If we have to keep the search text (I may be missing a case where we need it outside the method but for me it's not useful), we can probably keep the optimisation doing something like (if we don't need to store the search text, we can remove the if condition and always set m_Tree = m_TreeFull):
```c++
void display(const QString &newText) {
   if(newText.trimmed().length() < m_searchText.length()) m_Tree = m_TreeFull;
   if(!newText.trimmed().empty()) std::erase(std::remove_if( some code with m_Tree )); // I agree with the fact it's useless here to  compute for nothing ;)
   m_searchText = newText.trimmed();
   ... // the remaining code is good
}
```

-> this way, you don't manually iterate in the tree and don't have to handle the indices.

Also, for the m_search tree, I think we can recompute it when changing because we don't empty the text field when changing tab (from the search to another)?

The const probably won't make it faster but it helps for developers (we know that the string will not be updated in this method). Using a const reference will avoid to have to recreate a new QString